### DISCUSSION TITLE
Add multiple objects to the scene using AssetLoader

### DISCUSSION THREAD

**Discussion by ttyomaa6** [google/filament#5772] [Q&A]
I can pass 2 objects to the scene using two different instances of the AssetLoader class, one of them is the main one, and the rest of the models should change among themselves. There are no problems with adding the first model to the scene, but when the second model is added due to calling the transformToUnitCube method, the camera assumes a position near the second model. To make it clearer, I have a model of a person and sneakers, and I need the camera to always be on the person, and the models of sneakers changed. As I understand it, I need to pass several objects to the AssetLoader before adding them to the scene, I thought that the createInstancedAsset() method, but I can't understand until the end, I hope for early help)

**Comment by ttyomaa6** [comment]
I have moved on and now I have a new problem, my models are transformed in different ratios in the transformToUnitCube() method of the class, how to make their proportions after transformation preserved.

**Reply by romainguy** [reply]
Transform to unit cube does what it says: it ensures the model fits in a 1x1x1 bounding box. If you don't want this behavior you should simply not use that feature.

**Reply by ttyomaa6** [reply]
I understand this, thank you, but I can ask you a couple of questions, I tried without the conversion function and realized that the difference between the converted model and the usual one is about 50 times, is the load on the device different in this case or do I not understand its operation correctly?

**Reply by ttyomaa6** [reply]
Now I have big problems with camera position and objects position
